test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,856 @@
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.titleOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						title: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.multipleAttrs = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						url: {
+							source: 'core/post-meta',
+							args: { key: 'url_custom_field' },
+						},
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+		await requestUtils.activateTheme( 'twentytwentyone' );
+		await requestUtils.deactivatePlugin( 'gutenberg-test-block-bindings' );
+	} );
+
+	test.describe( 'Template context', () => {
+		test.beforeEach( async ( { admin, editor } ) => {
+			await admin.visitSiteEditor( {
+				postId: 'emptytheme//index',
+				postType: 'wp_template',
+			} );
+			await editor.canvas.locator( 'body' ).click();
+			await editor.openDocumentSettingsSidebar();
+		} );
+
+		test.describe( 'Paragraph', () => {
+			test( 'Should show the value of the custom field', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( variables.blocks.paragraph );
+				const paragraphBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Paragraph',
+				} );
+				const paragraphContent = await paragraphBlock.textContent();
+				expect( paragraphContent ).toBe(
+					variables.customFields.textKey
+				);
+			} );
+
+			test( 'Should lock the appropriate controls', async ( {
+				editor,
+				page,
+			} ) => {
+				await editor.insertBlock( variables.blocks.paragraph );
+				const paragraphBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Paragraph',
+				} );
+				await paragraphBlock.click();
+
+				// Alignment controls exist.
+				await expect(
+					page.getByRole( 'button', {
+						name: variables.labels.align,
+					} )
+				).toBeVisible();
+
+				// Format controls don't exist.
+				await expect(
+					page.getByRole( 'button', {
+						name: variables.labels.bold,
+					} )
+				).toBeHidden();
+
+				// Paragraph is not editable.
+				const isContentEditable =
+					await paragraphBlock.getAttribute( 'contenteditable' );
+				expect( isContentEditable ).toBe( 'false' );
```

Comment:
We have `toHaveAttribute` and also [`toBeEditable`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-editable).

Reply:
I've used `toHaveAttribute` this commit as part of a follow-up pull request: [link](https://github.com/WordPress/gutenberg/commit/3ddb71c9f834a09b9a1b3dc1bd81066256c4f3fe).

I tried using `toBeEditable`, but it seems it is always `true` even if `contenteditable` is set to false.